src/libcore/bitmap.cpp

```diff
@@ -185,11 +197,11 @@ void Bitmap::rebuild_struct(size_t channel_count) {
     m_struct = new Struct();
     for (auto ch: channels) {
         uint32_t flags = +Struct::Flags::None;
-        if (ch != "A" && ch != "W" && m_srgb_gamma)
+        if ((ch != "A" || m_pixel_format == PixelFormat::MultiChannel) && ch != "W" && m_srgb_gamma)
```

Comment:
I don't understand this logic here, and it seems very repeated across lines. can you clarify/refactor?

Reply:
With the new logic for creating multichannel Bitmaps by specifying the field names, if the Bitmap contains a field "A", it will be flagged as the alpha channel. Imagine now you have a multichannel bitmap with fields `[R, G, B, A, im.R, im.G, im.B, im.A]`.

When calling `split` on this Bitmap, a `StructConverter` will be created, taking as argument the "main" Bitmap, and the split one we're currently creating. For the `im` Bitmap, the field `im.A` will be flagged as alpha (as it should). However the `StructConverter` will raise an error because the alpha channels in the two structs are mismatched (`A` vs `im.A`).

This was not an issue before because we usually call `film.bitmap().split()` and the `HDRFilm::Bitmap` creates a multichannel bitmap (in case it has aovs) with generic channel names and renames them afterwards, hence not flagging any channel as the alpha in the multichannel bitmap. 

This modification to `rebuild_struct` aims at working around that, by simply not flagging any channel as alpha in the multichannel case, since the notion of alpha in this case is ambiguous and we don't do anything alpha related with it anyway. 

The same issue arises for the `W` channel. For now, the workaround for it is to fail if a channel called W is encountered in the `split` method, as the only time this channel should be added is manually in the `Integrator::render` method.

Reply:
Ok, that sounds reasonable to me. Can you refactor the code to move some conditionals at the top of the loop? (e.g. `bool is_alpha = ch == "A" && m_pixel_format != PixelFormat::MultiChannel;`. Please also add this explanation to the commit message.

Reply:
Sure. Do you also agree with the current handling of the "W" case?

Reply:
it's been too long since I wrote that code, so I probably don't have the same level of insight that you hold now. As far as I can remember, we can have multichannel images with a weight channel, and it's important to keep track of that and at some point divide by the weight as the image is exposed. This is the standard situation when rendering AOVs I suppose. As long as that is working, I don't see an issue.

Reply:
Regarding the `W` channel, I think it is fair to require the weight channel to already be applied before calling `split`, which is what we have at the moment. Therefore I am in favor of Baptiste's changes where the `split` method fails when encountering a `W` channel.